examples/hybrid/staggered_nonhydrostatic_model.jl

```diff
@@ -132,6 +132,90 @@ function default_cache(Y, params, numerics, simulation)
     )
 end
 
+
+function implicit_tendency!(Yₜ, Y, p, t)
+    (; apply_moisture_filter) = p
+    apply_moisture_filter && affect_filter!(Y)
+    @nvtx "implicit tendency" color = colorant"yellow" begin
+        ᶜρ = Y.c.ρ
+        ᶜuₕ = Y.c.uₕ
+        ᶠw = Y.f.w
+        (; ᶜK, ᶜΦ, ᶜts, ᶜp, params, ᶠupwind_product) = p
+        thermo_params = CAP.thermodynamics_params(params)
+        # Used for automatically computing the Jacobian ∂Yₜ/∂Y. Currently requires
+        # allocation because the cache is stored separately from Y, which means that
+        # similar(Y, <:Dual) doesn't allocate an appropriate cache for computing Yₜ.
+        if eltype(Y) <: Dual
+            ᶜK = similar(ᶜρ)
+            ᶜts = similar(ᶜρ, eltype(ᶜts).name.wrapper{eltype(ᶜρ)})
+            ᶜp = similar(ᶜρ)
+        end
```

Comment:
This appears to be causing the problem: `@code_warntype` tells me it can't infer the types of `ᶜp`.

@dennisYatunin is this still being used?